lib/px/utils/pxclient.lua

```diff
@@ -200,55 +200,101 @@ function M.load(config_file)
             end
         end
 
-        httpc:proxy_response(httpc:proxy_request())
+        local res, err = httpc:proxy_request()
+
+        if err or res.status >= 400 then
+            return false
+        end
+
+        httpc:proxy_response(res)
         httpc:set_keepalive()
+
+        ngx.exit(ngx.status)
+        return true
+    end
+
+    -- inteneral function, handles first party response that failed/bad status
+    -- return true for handled request
+    function default_response(content_type, content)
+        px_logger.error('Rendering default reponse on route ' .. ngx.var.uri .. 'content type: ' .. content_type .. 'body' .. content)
```

Comment:
why are you logging this as an error? don't this have the potential of spamming the customer's log?

Reply:
👍 
I think this might be important but we can make it a debug